rllib/utils/typing.py

```diff
@@ -13,14 +13,21 @@
     Union,
 )
 
-import numpy as np
 import gymnasium as gym
 
 from ray.rllib.utils.annotations import OldAPIStack
 
 if TYPE_CHECKING:
-    from ray.rllib.core.rl_module.rl_module import RLModuleSpec
+    # Modules might be missing but supply users with type hints if they are installed.
+    import jax.numpy as jnp
+    import keras
+    import tensorflow as tf
+    import torch
+    from numpy.typing import NDArray
+    from torch._prims_common import DeviceLikeType
```

Comment:
Same here, I'd avoid importing private APIs.

DeviceLikeType is just an [alias](https://github.com/pytorch/pytorch/blob/main/torch/_prims_common/__init__.py#L67), would make sense to implement it here, improving the `DeviceType`. I think you suggested it in your PR description.

`DeviceLike: TypeAlias = Union[str, torch.device, int]`